src/jit/rangecheck.cpp

```diff
@@ -1050,6 +1050,15 @@ bool RangeCheck::ComputeDoesOverflow(BasicBlock* block, GenTree* expr)
     {
         overflows = DoesPhiOverflow(block, expr);
     }
+    else if (expr->OperGet() == GT_CAST)
+    {
+        GenTreeCast* castExpr = expr->AsCast();
+        if (castExpr->CastToType() == TYP_UBYTE && castExpr->CastFromType() == TYP_INT)
+        {
+            overflows = false;
```

Comment:
Not sure here. Below I fixed `ComputeRange` to return 0-255 for the test case (it now respects GT_CAST) but then [here](https://github.com/dotnet/coreclr/blob/6b130c22db1c309a75dd3b679f04035e2c8fec9d/src/jit/rangecheck.cpp#L273) it used to stop due to "overflow".